src/vm/jitinterface.cpp

```diff
@@ -12570,8 +12572,16 @@ CorJitResult CallCompileMethodWithSEHWrapper(EEJitManager *jitMgr,
         flags.Set(CORJIT_FLAGS::CORJIT_FLAG_FRAMED);
     if (g_pConfig->JitAlignLoops())
         flags.Set(CORJIT_FLAGS::CORJIT_FLAG_ALIGN_LOOPS);
-    if (ftn->IsVersionableWithJumpStamp() || g_pConfig->AddRejitNops())
+
+    // Only set CORJIT_FLAG_PROF_REJIT_NOPS when we aren't versionable using 
+    // the Tiering mechanism and we are versionable using a jump stamp.
+    //
+    if ((!ftn->IsVersionableWithoutJumpStamp() && ftn->IsVersionableWithJumpStamp())
```

Comment:
This logic should rather be in `CORJIT_FLAGS GetCompileFlags` since it is specific to JIT.

Once you do that, it can be `ftn->IsVersionableWithJumpStamp()` again.